internal/controllers/catalog_test.go

```diff
@@ -0,0 +1,97 @@
+package controllers_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/operator-framework/operator-controller/internal/controllers"
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+	operatorsv1alpha1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ = Describe("Operator Controller Catalog Reconciler", func() {
```

Comment:
WDYT about making this whole section the `SetupWithManager` tests. So we're essentially making this the place that we test our watches (and anything else we might setup).
```suggestion
var _ = Describe("SetupWithManager", func() {
```